src/aiida/tools/archive/create.py

```diff
@@ -267,12 +267,13 @@ def querybuilder():
 
     if test_run:
         EXPORT_LOGGER.report('Test Run: Stopping before archive creation')
-        keys = set(
-            orm.Node.get_collection(backend).iter_repo_keys(
-                filters={'id': {'in': list(entity_ids[EntityTypes.NODE])}}, batch_size=batch_size
+        if node_ids := list(entity_ids[EntityTypes.NODE]):
```

Comment:
If this is a separate fix as you mention, could you please separate this out in an individual commit (no need for a separate PR) and add a test?